Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -230,19 +268,44 @@ function addAttributeToRenderResources(
     renderResources.featureIdVertexAttributeSetIndex = setIndex + 1;
   }
 
+  // The position attribute should always be in the first index.
+  const isPositionAttribute = semantic === VertexAttributeSemantic.POSITION;
+  const index = isPositionAttribute ? 0 : attributeIndex;
+  const componentsPerAttribute = AttributeType.getNumberOfComponents(type);
+
   const vertexAttribute = {
-    index: attributeIndex,
+    index: index,
     value: defined(attribute.buffer) ? undefined : attribute.constant,
     vertexBuffer: attribute.buffer,
     count: attribute.count,
-    componentsPerAttribute: AttributeType.getNumberOfComponents(type),
+    componentsPerAttribute: componentsPerAttribute,
     componentDatatype: componentDatatype,
     offsetInBytes: attribute.byteOffset,
     strideInBytes: attribute.byteStride,
     normalize: attribute.normalized,
   };
 
   renderResources.attributes.push(vertexAttribute);
+
+  if (!isPositionAttribute || !use2D) {
+    return;
+  }
+
+  // Add an additional attribute for the projected positions in 2D / CV.
+  const buffer2D = renderResources.runtimePrimitive.positionBuffer2D;
+  const positionAttribute2D = {
+    index: attributeIndex,
+    value: buffer2D ? undefined : attribute.constant,
```

Comment:
Is there ever a case where buffer2D won't be defined here?

Reply:
No, I guess it doesn't make sense for it to be